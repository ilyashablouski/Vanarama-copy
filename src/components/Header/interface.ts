export interface IHeroProps {
  flagText?: string;
}
