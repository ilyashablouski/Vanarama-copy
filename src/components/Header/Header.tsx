/* eslint-disable @typescript-eslint/no-unused-expressions */
/* eslint-disable import/no-cycle */
import React, { FC, memo, useState, useEffect } from 'react';
import { useRouter } from 'next/router';
import dynamic from 'next/dynamic';
import { LazyLoadComponent } from 'react-lazy-load-image-component';
import cx from 'classnames';
import localForage from 'localforage';
import { IBaseProps } from '@vanarama/uibook/lib/interfaces/base';
import Logo from '@vanarama/uibook/lib/components/atoms/logo';
import { ILinkProps } from '../RouterLink/interface';
import RouterLink from '../RouterLink/RouterLink';
// import HeaderMenu from './HeaderMenu';
import {
  GetPerson_getPerson as Person,
  GetPerson,
} from '../../../generated/GetPerson';
import { useMobileViewport } from '../../hooks/useMediaQuery';

const HeaderMenu = dynamic(() => import('./HeaderMenu'));
const Button = dynamic(() =>
  import('@vanarama/uibook/lib/components/atoms/button'),
);
const Icon = dynamic(
  () => import('@vanarama/uibook/lib/components/atoms/icon'),
  {
    ssr: false,
  },
);
const SearchCircle = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/SearchOutline'),
  {
    ssr: false,
  },
);
const PersonCircleSharp = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/PersonCircleSharp'),
  {
    ssr: false,
  },
);
const HomeOutline = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/HomeOutline'),
  {
    ssr: false,
  },
);
const ReceiptOutline = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/ReceiptOutline'),
  {
    ssr: false,
  },
);
const CarOutline = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/CarOutline'),
  {
    ssr: false,
  },
);
const Menu = dynamic(() => import('@vanarama/uibook/lib/assets/icons/Menu'), {
  ssr: false,
});
const LogOutOutline = dynamic(
  () => import('@vanarama/uibook/lib/assets/icons/LogOutOutline'),
  {
    ssr: false,
  },
);
const Close = dynamic(() => import('@vanarama/uibook/lib/assets/icons/Close'), {
  ssr: false,
});
const Call = dynamic(() => import('@vanarama/uibook/lib/assets/icons/Call'), {
  ssr: false,
});

export interface IHeaderPromoImage {
  legacyUrl?: string;
  url: string;
  image: {
    fileName: string;
    url: string;
  };
}
export interface IHeaderLink extends ILinkProps {
  id?: string;
  as?: string;
  highlight?: boolean;
  promotionalImage?: IHeaderPromoImage;
}

export interface IHeaderProps extends IBaseProps {
  topBarLinks: IHeaderLink[];
  loginLink: ILinkProps;
  phoneNumberLink: ILinkProps;
  onLogOut: () => void;
}

export const Header: FC<IHeaderProps> = memo(props => {
  const router = useRouter();
  const {
    className,
    topBarLinks,
    loginLink,
    phoneNumberLink,
    onLogOut,
  } = props;
  const [person, setPerson] = useState<Person | null>(null);
  const [ordersLength, setOrdersLength] = useState<number | null>(null);
  const [quotesLength, setQuotesLength] = useState<number | null>(null);
  const [isMenuOpen, setOpenMenu] = useState(false);
  const [isMyAccountOpen, setOpenMyAccount] = useState(false);

  useEffect(() => {
    if (!person) {
      localForage.getItem('person').then(value => {
        if ((value as GetPerson)?.getPerson)
          setPerson((value as GetPerson)?.getPerson as Person);
      });
    }
    if (!ordersLength) {
      localForage.getItem('ordersLength').then(value => {
        if (value) setOrdersLength(value as number);
      });
    }
    if (!quotesLength) {
      localForage.getItem('quotesLength').then(value => {
        if (value) setQuotesLength(value as number);
      });
    }
    setOpenMyAccount(false);
  }, [person, ordersLength, quotesLength, router]);

  useEffect(() => {
    setOpenMenu(false);
  }, [router]);

  return (
    <header
      style={{ position: useMobileViewport() ? 'fixed' : 'relative' }}
      className={cx('header', className)}
      data-testid="header"
    >
      {' '}
      <div className="header-content">
        {' '}
        <RouterLink
          link={{ href: '/', label: '' }}
          className="logo header-logo"
          classNames={{ color: 'orange', plain: true }}
        >
          {' '}
          <LazyLoadComponent visibleByDefault={typeof window === 'undefined'}>
            <Logo asset="vanarama" />{' '}
          </LazyLoadComponent>
        </RouterLink>{' '}
        {!useMobileViewport() && (
          <label className="header-search" htmlFor="search">
            {' '}
            {/* {TODO: commit for this search lines should be reverted after implement search functionality} */}
            <Icon icon={<SearchCircle />} />{' '}
            <input
              className="header-search--input"
              id="search"
              type="text"
              disabled
              placeholder=""
            />{' '}
            {/* <div className="header-search--results -is-hidden" /> */}{' '}
          </label>
        )}
        <RouterLink link={phoneNumberLink} className="header-tel">
          {' '}
          <Icon icon={<Call />} size="xsmall" /> <span>01442 838195</span>{' '}
        </RouterLink>{' '}
        <LazyLoadComponent visibleByDefault={typeof window === 'undefined'}>
          <div className="header-account">
            {' '}
            {person ? (
              <>
                {' '}
                <Button
                  withoutDefaultClass
                  fill="clear"
                  onClick={() => setOpenMyAccount(!isMyAccountOpen)}
                  className="header-account--toggle"
                  label={
                    <>
                      {' '}
                      <Icon icon={<PersonCircleSharp />} size="xsmall" />{' '}
                      <span>My Account</span>{' '}
                    </>
                  }
                />{' '}
                <div
                  className={cx('header-account--content', {
                    '-open': isMyAccountOpen,
                  })}
                >
                  {' '}
                  <div className="header-account--header">
                    {' '}
                    <span>
                      {' '}
                      Hi{' '}
                      {person?.firstName &&
                        person?.lastName &&
                        `, ${person?.firstName} ${person?.lastName}`}{' '}
                    </span>{' '}
                  </div>{' '}
                  <ul className="header-account--nav">
                    {' '}
                    <li>
                      {' '}
                      <RouterLink
                        className="header-account--link"
                        link={{
                          href: `/account/my-details/[uuid]`,
                          label: 'Dashboard',
                          query: { partyByUuid: person.partyUuid },
                        }}
                        as={`/account/my-details/${person.uuid}?partyByUuid=${person.partyUuid}`}
                      >
                        {' '}
                        <Icon icon={<HomeOutline />} size="xsmall" />{' '}
                        <span>Dashboard</span>{' '}
                      </RouterLink>{' '}
                    </li>{' '}
                    <li>
                      {' '}
                      <RouterLink
                        className="header-account--link"
                        link={{
                          href: quotesLength
                            ? '/account/my-quotes/[partyByUuid]'
                            : `/account/my-details/[uuid]`,
                          label: 'My Quotes',
                          query: {
                            partyByUuid: person.partyUuid,
                            uuid: person.uuid,
                          },
                        }}
                        as={
                          quotesLength
                            ? `/account/my-quotes/${person.partyUuid}?uuid=${person.uuid}`
                            : `/account/my-details/${person.uuid}?partyByUuid=${person.partyUuid}`
                        }
                      >
                        {' '}
                        <Icon icon={<ReceiptOutline />} size="xsmall" />{' '}
                        <span>My Quotes</span>{' '}
                      </RouterLink>{' '}
                    </li>{' '}
                    <li>
                      {' '}
                      <RouterLink
                        className="header-account--link"
                        link={{
                          href: ordersLength
                            ? '/account/my-orders/[partyByUuid]'
                            : `/account/my-details/[uuid]`,
                          label: 'My Orders',
                          query: {
                            partyByUuid: person.partyUuid,
                            uuid: person.uuid,
                          },
                        }}
                        as={
                          ordersLength
                            ? `/account/my-orders/${person.partyUuid}?uuid=${person.uuid}`
                            : `/account/my-details/${person.uuid}?partyByUuid=${person.partyUuid}`
                        }
                      >
                        {' '}
                        <Icon icon={<CarOutline />} size="xsmall" />{' '}
                        <span>My Orders</span>{' '}
                      </RouterLink>{' '}
                    </li>{' '}
                    <li>
                      {' '}
                      <RouterLink
                        className="header-account--link"
                        link={{ href: router.pathname, label: 'Log Out' }}
                        as={router.asPath}
                        onClick={async () => {
                          await onLogOut();
                          setPerson(null);
                        }}
                        replace
                      >
                        {' '}
                        <Icon icon={<LogOutOutline />} size="xsmall" />{' '}
                        <span>Log Out</span>{' '}
                      </RouterLink>{' '}
                    </li>{' '}
                  </ul>{' '}
                </div>{' '}
              </>
            ) : (
              <Button
                className="header-account--toggle"
                fill="clear"
                label={
                  <RouterLink link={loginLink}>
                    {' '}
                    <span>Login / Register</span>{' '}
                  </RouterLink>
                }
              />
            )}{' '}
          </div>{' '}
        </LazyLoadComponent>
        <HeaderMenu
          menuLinks={topBarLinks}
          open={isMenuOpen}
          onClickMenu={() => setOpenMenu(false)}
        />{' '}
        <Button
          className={cx('header-navtoggle', { '-open': isMenuOpen })}
          onClick={() => setOpenMenu(!isMenuOpen)}
          withoutDefaultClass
          label={
            <>
              {' '}
              <Icon icon={<Menu />} color="darker" />{' '}
              <Icon icon={<Close />} color="darker" />{' '}
            </>
          }
          size="expand"
          color="inherit"
          fill="clear"
        />{' '}
      </div>{' '}
    </header>
  );
});

export default Header;
