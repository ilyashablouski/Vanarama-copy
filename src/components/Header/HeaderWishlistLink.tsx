import React from 'react';

import Icon from 'core/atoms/icon';
import Heart from 'core/assets/icons/Heart';

import useWishlist from '../../hooks/useWishlist';

import RouterLink from '../RouterLink';

function HeaderWishlistLink() {
  const { wishlistVehicles } = useWishlist();

  return (
    <RouterLink
      className="header-wishlist"
      link={{ href: '/wishlist', label: '' }}
    >
      <Icon icon={<Heart />} size="large" />
      {wishlistVehicles.length > 0 && (
        <span className="badge">{wishlistVehicles.length}</span>
      )}
    </RouterLink>
  );
}

export default HeaderWishlistLink;
