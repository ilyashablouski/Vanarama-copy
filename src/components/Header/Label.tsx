import React from 'react';

interface ILableProps {
  text: string;
}
const Label: React.FC<ILableProps> = ({ text }) => (
  <span
    style={{
      color: 'white',
      backgroundColor: '#ec6409',
      fontSize: '11px',
      paddingRight: '8px',
      paddingLeft: '8px',
      paddingTop: '1px',
      paddingBottom: '1px',
      borderRadius: '4px',
      marginLeft: '10px',
      display: 'inline-block',
    }}
  >
    {text}
  </span>
);

export default React.memo(Label);
