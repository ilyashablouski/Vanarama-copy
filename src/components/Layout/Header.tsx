

export default () => (
  <header className="Header">
    This is your header
  </header>
);
