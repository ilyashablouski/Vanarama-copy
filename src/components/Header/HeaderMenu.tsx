/* eslint-disable import/no-cycle */
import React, { FC, memo } from 'react';
import dynamic from 'next/dynamic';
import cx from 'classnames';
import { IBaseProps } from 'core/interfaces/base';
import { IHeaderLink } from './Header';
import Skeleton from '../Skeleton';

const HeaderMenuLink = dynamic(() => import('./HeaderMenuLink'));
const Button = dynamic(() => import('core/atoms/button'), {
  loading: () => <Skeleton count={1} />,
});

export interface IHeaderMenuProps extends IBaseProps {
  menuLinks: IHeaderLink[];
  open: boolean;
  onClickMenu: () => void;
}

const HeaderMenu: FC<IHeaderMenuProps> = memo(props => {
  const { menuLinks, open, onClickMenu } = props;

  return (
    <nav
      id="nav"
      className={cx('menu', {
        '-open': open,
      })}
    >
      <ul className="menu-primary">
        <li className="menu-li menu-title">
          <Button
            withoutDefaultClass
            className="link"
            onClick={() => onClickMenu()}
            dataTestId="menu-title"
            color="black"
            fill="clear"
            label="Menu"
          />
        </li>
        {!!menuLinks.length &&
          menuLinks.map((entry, idx) => (
            <HeaderMenuLink
              key={`${entry.id}_${idx.toString()}`}
              link={entry}
              isMenuOpen={open}
            />
          ))}
      </ul>
    </nav>
  );
});

export default HeaderMenu;
