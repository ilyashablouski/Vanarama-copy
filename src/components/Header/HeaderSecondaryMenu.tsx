/* eslint-disable import/no-cycle */
import React, { FC, memo, useState, useEffect, useMemo } from 'react';
import dynamic from 'next/dynamic';
import { useRouter } from 'next/router';
import cx from 'classnames';
import { IBaseProps } from 'core/interfaces/base';
import { LazyLoadComponent } from 'react-lazy-load-image-component';
import RouterLink from '../RouterLink/RouterLink';
import { IHeaderLink, IHeaderPromoImage } from './Header';
import Skeleton from '../Skeleton';
import Label from './Label';
import { isServerRenderOrAppleDevice } from '../../utils/deviceType';

const Image = dynamic(() => import('core/atoms/image'), {
  loading: () => <Skeleton count={4} />,
});
const Button = dynamic(() => import('core/atoms/button'), {
  loading: () => <Skeleton count={1} />,
});

const Icon = dynamic(() => import('core/atoms/icon'), {
  ssr: false,
});
const FlameSharp = dynamic(() => import('core/assets/icons/FlameSharp'), {
  ssr: false,
});

export interface IHeaderSecondaryMenuProps extends IBaseProps {
  links: IHeaderLink[];
  title: string;
  onClickTitle: () => void;
  isTabletOrMobile: boolean;
  isMenuOpen: boolean;
  isSecondaryMenuOpen: boolean;
  promotionalImage?: IHeaderPromoImage;
}

const HeaderSecondaryMenu: FC<IHeaderSecondaryMenuProps> = memo(props => {
  const router = useRouter();
  const {
    links,
    title,
    onClickTitle,
    isTabletOrMobile,
    isSecondaryMenuOpen,
    isMenuOpen,
    promotionalImage,
  } = props;
  const firstChildrenLinks: IHeaderLink | undefined = useMemo(
    () => links.find(el => !!el.children?.length && !el.promotionalImage?.url),
    [links],
  );

  const tertiaryLinks: IHeaderLink[] = useMemo(
    () => links.filter(link => !!link.children?.length),
    [links],
  );

  const [activeTertiaryMenu, setActiveTertiaryMenu] = useState<string | null>(
    firstChildrenLinks?.id || '',
  );

  useEffect(() => {
    if (isTabletOrMobile) {
      setActiveTertiaryMenu(null);
    } else {
      setActiveTertiaryMenu(firstChildrenLinks?.id || '');
    }
  }, [
    router,
    isMenuOpen,
    isSecondaryMenuOpen,
    isTabletOrMobile,
    setActiveTertiaryMenu,
    firstChildrenLinks?.id,
  ]);

  const linkClassName = (classes: {
    title?: boolean;
    open?: boolean;
    highlight?: boolean;
    half?: boolean;
    withChildren?: boolean;
  }) =>
    cx('menu-li', {
      'menu-title': classes.title,
      '-open': classes.open,
      '-highlight': classes.highlight,
      '-span-half': classes.half,
      '-with-children': classes.withChildren,
    });

  return (
    <div
      className="menu-secondary--wrapper-outer"
      data-testid="menu-secondary--wrapper-outer"
    >
      <LazyLoadComponent visibleByDefault={isServerRenderOrAppleDevice}>
        <div className="menu-secondary--wrapper-inner">
          <ul className="menu-secondary">
            <li className={linkClassName({ title: true })}>
              <Button
                withoutDefaultClass
                className="link"
                onClick={el => {
                  el.preventDefault();
                  onClickTitle();
                }}
                dataTestId="menu-title"
                color="black"
                fill="clear"
                label={title}
              />
            </li>
            {links.map((link: IHeaderLink) => (
              <li
                key={`${link.label}_${title}`}
                className={linkClassName({
                  open:
                    !!link.children?.length && activeTertiaryMenu === link.id,
                  highlight: link.highlight,
                  withChildren: !!link.children?.length,
                })}
                data-testid={link.label}
              >
                {!link.href ? (
                  <>
                    <Button
                      key={`${link.label}_${title}`}
                      withoutDefaultClass
                      className={
                        link.highlight ? 'link -white' : 'link -inherit'
                      }
                      color="black"
                      fill="clear"
                      size="initial-size"
                      label={
                        link.highlightLabel ? link.highlightLabel : link.label
                      }
                      onClick={
                        isTabletOrMobile && link.children?.length
                          ? () => {
                              setActiveTertiaryMenu(link?.id || '');
                            }
                          : undefined
                      }
                      onMouseOver={
                        !isTabletOrMobile && link.children?.length
                          ? () => {
                              setActiveTertiaryMenu(link?.id || '');
                            }
                          : undefined
                      }
                      onFocus={
                        link.children?.length
                          ? () => {
                              setActiveTertiaryMenu(link?.id || '');
                            }
                          : undefined
                      }
                    >
                      {link.highlight && (
                        <Icon
                          icon={<FlameSharp />}
                          color="white"
                          size="xsmall"
                        />
                      )}
                    </Button>
                  </>
                ) : (
                  <RouterLink
                    link={link}
                    classNames={{
                      color: (!!link.highlight && 'white') || 'inherit',
                    }}
                    onClick={
                      isTabletOrMobile && link.children?.length
                        ? el => {
                            el.preventDefault();
                            setActiveTertiaryMenu(link.id || null);
                          }
                        : undefined
                    }
                    onMouseOver={
                      !isTabletOrMobile && link.children?.length
                        ? () => {
                            setActiveTertiaryMenu(link?.id || '');
                          }
                        : undefined
                    }
                    onFocus={
                      link.children?.length
                        ? () => {
                            setActiveTertiaryMenu(link?.id || '');
                          }
                        : undefined
                    }
                  >
                    {link.highlight && (
                      <Icon icon={<FlameSharp />} color="white" size="xsmall" />
                    )}
                    <span>{link.label}</span>
                    {link.highlightLabel && (
                      <Label text={link.highlightLabel} />
                    )}
                  </RouterLink>
                )}
              </li>
            ))}
          </ul>
          {tertiaryLinks.map(tertiaryBlock => (
            <React.Fragment key={`menu-tertiary-${tertiaryBlock?.id}`}>
              <ul
                className={cx('menu-tertiary', {
                  '-open': activeTertiaryMenu === tertiaryBlock.id,
                })}
              >
                <li className={linkClassName({ title: true })}>
                  <Button
                    withoutDefaultClass
                    className="link"
                    onClick={el => {
                      el.preventDefault();
                      setActiveTertiaryMenu(null);
                    }}
                    dataTestId={`menu-tertiary-${tertiaryBlock.id}`}
                    color="black"
                    fill="clear"
                    label={tertiaryBlock.label}
                  />
                </li>
                {(tertiaryBlock.children as IHeaderLink[]).map(
                  (linkSecondary: IHeaderLink) => (
                    <li
                      key={linkSecondary.label}
                      className={linkClassName({
                        highlight: linkSecondary.highlight,
                        half: tertiaryBlock?.children!.length > 4,
                      })}
                    >
                      <RouterLink link={linkSecondary} as={linkSecondary.as}>
                        {linkSecondary.highlight && (
                          <Icon
                            icon={<FlameSharp />}
                            color="white"
                            size="xsmall"
                          />
                        )}
                        <span>{linkSecondary.label}</span>
                      </RouterLink>
                    </li>
                  ),
                )}
              </ul>
              {tertiaryBlock.promotionalImage?.url && (
                <div
                  className={cx('menu-featured', 'tertiary', {
                    '-hide':
                      activeTertiaryMenu !== tertiaryBlock.id &&
                      activeTertiaryMenu !== '',
                  })}
                >
                  <RouterLink
                    link={{
                      href: tertiaryBlock.promotionalImage?.url,
                      label: '',
                    }}
                  >
                    <Image
                      optimisedHost={process.env.IMG_OPTIMISATION_HOST}
                      optimisationOptions={{
                        quality: 40,
                      }}
                      src={
                        tertiaryBlock.promotionalImage?.image.url ||
                        '/img-placeholder.png'
                      }
                      alt={tertiaryBlock.promotionalImage?.image.fileName}
                    />
                  </RouterLink>
                </div>
              )}
            </React.Fragment>
          ))}

          {promotionalImage?.url && (
            <div className="menu-featured">
              <RouterLink link={{ href: promotionalImage?.url, label: '' }}>
                <Image
                  optimisedHost={process.env.IMG_OPTIMISATION_HOST}
                  optimisationOptions={{
                    quality: 40,
                  }}
                  src={promotionalImage?.image.url || '/img-placeholder.png'}
                  alt={promotionalImage?.image.fileName}
                />
              </RouterLink>
            </div>
          )}
        </div>
      </LazyLoadComponent>
    </div>
  );
});

export default HeaderSecondaryMenu;
