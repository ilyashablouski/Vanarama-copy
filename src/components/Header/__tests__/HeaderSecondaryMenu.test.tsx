import renderer from 'react-test-renderer';
import React from 'react';
import preloadAll from 'jest-next-dynamic';
import { render } from '@testing-library/react';
import { TOP_BAR_LINKS } from '../../../models/enum/HeaderLinks';
import HeaderSecondaryMenu from '../HeaderSecondaryMenu';
import { IHeaderLink } from '../Header';

const mocks = {
  links: TOP_BAR_LINKS[1].children as IHeaderLink[],
  title: 'title',
  onClickTitle: jest.fn(),
  isMobile: false,
  isMenuOpen: false,
  promotionalImage: {
    url: '',
    image: {
      url: '',
      fileName: '',
    },
  },
};

jest.mock('../../../hooks/useMediaQuery');
jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/',
  }),
}));

describe('<HeaderSecondaryMenu />', () => {
  beforeEach(async () => {
    await preloadAll();
  });
  beforeEach(() => {
    jest.clearAllMocks();
    render(<HeaderSecondaryMenu {...mocks} />);
  });

  it('renders correctly', () => {
    const getComponent = () => {
      return renderer.create(<HeaderSecondaryMenu {...mocks} />).toJSON();
    };
    const tree = getComponent();
    expect(tree).toMatchSnapshot();
  });
});
