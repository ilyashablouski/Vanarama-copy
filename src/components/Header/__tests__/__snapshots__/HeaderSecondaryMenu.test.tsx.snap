// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderSecondaryMenu /> renders correctly 1`] = `
<div
  className="menu-secondary--wrapper-outer"
  data-testid="menu-secondary--wrapper-outer"
>
  <div
    className="menu-secondary--wrapper-inner"
  >
    <ul
      className="menu-secondary"
    >
      <li
        className="menu-li menu-title"
      >
        <button
          className="link -black -regular -clear"
          data-testid="menu-title"
          onClick={[Function]}
        >
          <div
            className="button--inner"
          >
            title
          </div>
        </button>
      </li>
      <li
        className="menu-li -highlight"
        data-testid="VANS Special Offers"
      >
        <button
          className="link -white -black -initial-size -clear"
        >
          <div
            className="button--inner"
          >
            VANS Special Offers
          </div>
        </button>
      </li>
      <li
        className="menu-li -open -with-children"
        data-testid="Vans By Manufacturer"
      >
        <button
          className="link -inherit -black -initial-size -clear"
          onFocus={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Manufacturer
          </div>
        </button>
      </li>
      <li
        className="menu-li -with-children"
        data-testid="Vans By Size"
      >
        <button
          className="link -inherit -black -initial-size -clear"
          onFocus={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Size
          </div>
        </button>
      </li>
      <li
        className="menu-li -with-children"
        data-testid="Vans By Type"
      >
        <button
          className="link -inherit -black -initial-size -clear"
          onFocus={[Function]}
          onMouseOver={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Type
          </div>
        </button>
      </li>
      <li
        className="menu-li"
        data-testid="Van Leasing Explained"
      >
        <a
          className="link -inherit"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/finance-options"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Van Leasing Explained
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="menu-tertiary -open"
    >
      <li
        className="menu-li menu-title"
      >
        <button
          className="link -black -regular -clear"
          data-testid="menu-tertiary-b21cfa5d-e57f-4d7e-867f-4a1870cfb879"
          onClick={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Manufacturer
          </div>
        </button>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Citroen"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Citroen
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Ford"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Ford
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Benz"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Benz
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Nissan"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Nissan
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Renault"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Renault
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Vauxhall"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Vauxhall
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Fiat"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Fiat
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Isuzu"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Isuzu
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Mitsubishi"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Mitsubishi
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Peugeot"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Peugeot
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Toyota"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Toyota
          </span>
        </a>
      </li>
      <li
        className="menu-li -span-half"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing/Volkswagen"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Volkswagen
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="menu-tertiary"
    >
      <li
        className="menu-li menu-title"
      >
        <button
          className="link -black -regular -clear"
          data-testid="menu-tertiary-43df3e0b-f248-457d-bf3c-ec1fe4def7e3"
          onClick={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Size
          </div>
        </button>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Small%20Van"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Small
          </span>
        </a>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Medium%20Van"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Medium
          </span>
        </a>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Large%20Van"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Large
          </span>
        </a>
      </li>
    </ul>
    <ul
      className="menu-tertiary"
    >
      <li
        className="menu-li menu-title"
      >
        <button
          className="link -black -regular -clear"
          data-testid="menu-tertiary-44c0f32b-f034-4b84-ac49-0c2b76140c96"
          onClick={[Function]}
        >
          <div
            className="button--inner"
          >
            Vans By Type
          </div>
        </button>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Refrigerated%20Van"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Refrigerated
          </span>
        </a>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Dropside%20Tipper,Luton%20Box%20Van"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Tippers/Lutons
          </span>
        </a>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Minibus"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Crew/Minibus
          </span>
        </a>
      </li>
      <li
        className="menu-li"
      >
        <a
          className="link"
          data-menu={null}
          data-testid="router-link"
          href="/van-leasing?bodyStyles=Specialist"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseOver={[Function]}
        >
          <span>
            Specialist
          </span>
        </a>
      </li>
    </ul>
    
  </div>
</div>
`;
